.ado/variables/msbuild.yml

```diff
@@ -1,7 +1,7 @@
 variables:
-  MSBuildArchitecture: x64
-  MSBuildPreferredToolArchitecture: x64
-  MSBuildPlatformToolset: v141
+  MSBuildArchitecture: x86
+  MSBuildPreferredToolArchitecture: x86
+  MSBuildPlatformToolset: v142
```

Comment:
why are we setting these? should just inherit them from whatever version we're using of VS

Setting these values here will add friction to move to VS202x

Reply:
This allows to override the value in the build system. This is also true for MSBuild properties such as `IntDir` or `GeneratedFilesDir`.

This is a common practice in CI workflows.